### PR TITLE
chore: consolidating path validation logic

### LINKED ISSUE #2091: chore(internal): consolidate path validation logic
## Description
Our current logic for path validation & converting to urls is quite scattered.  For example, the regression in https://github.com/GlareDB/glaredb/pull/2089 was caused by some incorrect url handling. 

We have many spots we are performing the same checking logic for dirs.
- https://github.com/GlareDB/glaredb/blob/833bd1fd7dcb1d215fb555ec54f44ad6005fe2ed/crates/testing/src/slt/runner.rs#L50
- https://github.com/GlareDB/glaredb/blob/833bd1fd7dcb1d215fb555ec54f44ad6005fe2ed/crates/metastore/src/util.rs#L42
- https://github.com/GlareDB/glaredb/blob/833bd1fd7dcb1d215fb555ec54f44ad6005fe2ed/crates/glaredb/src/commands.rs#L184
- https://github.com/GlareDB/glaredb/blob/833bd1fd7dcb1d215fb555ec54f44ad6005fe2ed/bindings/python/src/connect.rs#L164
- https://github.com/GlareDB/glaredb/blob/833bd1fd7dcb1d215fb555ec54f44ad6005fe2ed/bindings/nodejs/src/connection.rs#L274

### PR DESCRIPTION
**Pull request by seve-martinez**
Closes #2091.

Consolidated repetitive checks for existence of paths and creation of directories into a single function within the `ioutil` crate.

Added a dependency to each subcrate where this logic existed.

Added a new variant to the `MetastoreError` that converts from `io::Error`.

### CODE REVIEW

File: crates/testing/src/slt/runner.rs [hunk 1]
```diff
@@ -46,19 +47,7 @@ impl SltRunner {
...
-
-        if !dir.is_dir() {
-            return Err(anyhow!(
-                "Path `{}` is not a valid directory",
-                dir.to_string_lossy()
-            ));
-        }
-
-        // Get the absolute path
+        ensure_dir(dir)?;
```
Comment (universalmind303) [inline crates/testing/src/slt/runner.rs:50]: this one looks like it's not supposed to create the dirs. We can probably revert this.

Comment (seve-martinez) [reply crates/testing/src/slt/runner.rs]: Thanks for the catch. I've updated the commit 🚀

**Comment by universalmind303** [comment]
@seve-martinez this looks good overall. The ci does seem to be failing with some linting errors.  Can you fix these?

**Comment by seve-martinez** [comment]
> @seve-martinez this looks good overall. The ci does seem to be failing with some linting errors. Can you fix these?

Updated PR

**Review by universalmind303 — APPROVED** [review]
@scsmithr @greyscaled could one of you merge this? the integration tests are failing due to unrelated issues.

**Pull request merged**